### PR TITLE
Read Only Mode

### PR DESCRIPTION
**Pull request by farhanW3**
## PR-Codex overview
The focus of this PR is to introduce an `ENGINE_MODE` environment variable for sandbox mode enforcement and update related logic.

### Detailed summary
- Added `ENGINE_MODE` environment variable with options `sandbox` and `unrestricted`
- Skipped auth check in sandbox mode in `auth.ts`
- Implemented `withEnforceEngineMode` middleware for sandbox mode enforcement
- Updated `index.ts` to include `withEnforceEngineMode` middleware

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/server/middleware/auth.ts [hunk 1]
```diff
@@ -106,6 +106,9 @@ export const withAuth = async (server: FastifyInstance) => {
   // Note: in the onRequest hook, request.body will always be undefined, because the body parsing happens before the preValidation hook.
   // https://fastify.dev/docs/latest/Reference/Hooks/#onrequest
   server.addHook("preValidation", async (req, res) => {
+    if (env.READ_ONLY) {
```
Comment (arcoraven) [inline src/server/middleware/auth.ts:109]: suggestion:
```
ENGINE_MODE?: 'sandbox' | 'ecosystem'
```

File: src/server/middleware/logs.ts [hunk 2]
```diff
@@ -25,6 +26,14 @@ export const withRequestLogs = async (server: FastifyInstance) => {
         });
       }
     }
+
+    if (env.READ_ONLY && request.method !== "GET") {
```
Comment (arcoraven) [inline src/server/middleware/logs.ts:30]: if (env.ENGINE_MODE === 'sandbox'){
  await withEnforceSandbox(server);
}

File: src/server/middleware/auth.ts (hunk 1 above)
Comment (farhanW3) [reply src/server/middleware/auth.ts]: done

File: src/server/middleware/logs.ts (hunk 2 above)
Comment (farhanW3) [reply src/server/middleware/logs.ts]: done

**Review by arcoraven — APPROVED** [review]

**Pull request merged**